### PR TITLE
Update tailor module to use local storage

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

- Creates a single entry point to tailor module 'fetchData'
- Depending on arguments passed to fetchData it will either pull data from local storage or retrieve from tailor end point.
- Tailor module now saves data in local storage for 24 hours.
- Update the contributions-epic-regulars.js ab-test to use the updated tailor module
- Update the tailor-survey.js ab-test to use the updated tailor module

## What is the value of this and can you measure success?

Simplifies tailor module, separation of concerns (tailor module just gets/saves data, dependants handle data), saves calls to tailor if data saved in local storage.

## Does this affect other platforms - Amp, Apps, etc?

No

## Screenshots

N/A

## Tested in CODE?

No

### CODE REVIEW

File: static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js [hunk 1]
```diff
@@ -1,34 +1,95 @@
...
-    function getRegularStatus(browserID) {
-        return getSuggestions(browserID).then(function(res) {
-            try {
-                return res.userDataForClient.regular;
-            } catch (e) {
-                return false
-            }
-        })
+    function handleResponse(type, data) {
+        var day = 1000 * 60 * 60 * 24;
```
Comment (lindseydew) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: I think an hour will be sufficient here

File: static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js [hunk 2]
```diff
@@ -1,34 +1,95 @@
...
+
+        storage.local.set('gu.tailor.' + type, data, { expires: Date.now() + day });
```
Comment (lindseydew) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: Do we need to worry about checking the existence of local storage for a browser? Or is this handled in the storage module?

File: static/src/javascripts-legacy/projects/common/modules/experiments/tests/contributions-epic-regulars.js [hunk 3]
```diff
@@ -112,6 +110,16 @@ define([
         }
     }
 
+    function isRegular() {
```
Comment (lindseydew) [inline static/src/javascripts-legacy/projects/common/modules/experiments/tests/contributions-epic-regulars.js]: Nice!

File: static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js [hunk 4]
```diff
@@ -1,34 +1,95 @@
...
+    storage,
+    reportError
 ) {
-    function getEmail(browserId) {
-        return fetchJson('https://tailor.guardianapis.com/email/' + browserId +'?emailIds=39,3322,3039,1950,38,3698', {
-            method: 'get'
-        });
+
+    var URLS = {
+        suggestions: 'https://tailor.guardianapis.com/suggestions?browserId='
```
Comment (lindseydew) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: Would it be possible to configure the call to tailor behind a switch?

**Review by lindseydew — COMMENTED** [review]
Amazing! Thank you for this

File: static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js (hunk 2 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: This check is handled by the storage module

File: static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js (hunk 1 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: OK I'll update this now

File: static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js (hunk 4 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: Sure, so if it's off we return an empty object (unless we have data in local storage)...?

Comment (lindseydew) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: Yeah that sounds like the right behaviour. Clients need to be able to handle the empty response anyway, so they can be agnostic as to whether Tailor is on or off

File: static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js [hunk 5]
```diff
@@ -1,34 +1,100 @@
...
+    function fetchData(type, queryParams) {
+        var tailorData = storage.local.get('gu.tailor.' + type);
+
+        // if data in local storage return this
+        if (tailorData) {
+            return Promise.resolve(tailorData);
+        }
+
+        // if no browserId or use tailor switch is off then return an empty object
+        if (!browserId || !config.switches.useTailorEndpoints) {
```
Comment (lindseydew) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: Ace - thanks 👍

File: static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js (hunk 4 above)
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js]: Cool, I've just put this in!

**Comment by GHaberis** [comment]
@jranks123 I've updated this PR to include your new test

**Comment by GHaberis** [comment]
@guardian/dotcom-platform if anyone can spare a bit of time to take a look at this that'd be great! thanks!

**Review by jranks123 — APPROVED** [review]
Looks good to me. Let's keep an eye on it when you merge to ensure that service carries on as expected.

**Pull request merged**